### PR TITLE
Add true JSON and YAML display options for events

### PR DESCRIPTION
**Pull request by luiseeo**
I'm adding different options for displaying the events, aka "notations".

Original motivation: I got tired of looking at "qp-batch-summary" events, which are nested, and feeling lost.

So far, there's three notations:
- "Flat", this is the current/old implementation, I left it alone for the most part. It's pretty readable, except if the event are nested. It is also not friendly to any automated parsing.
- "JSON", json complaint format, for some events it's quite unreadable (the one with strings with multiple lines), but this could be a good option if you want to export events and want to process them in a separate application.
- "YAML", yaml complaint format. It is very readable including nested events. And could still be read by automated parsing.

The notations have 2 forms: SingleLine and Free. Singleline is the one used by the main window, where only the first line is visible. Free can use as many newlines as wanted and it's used by the details window. I found that creating functions for each of these forms was easier than trying to special-case a lot of places in the formatting logic.

For future: The functions in QJsonUtils are begging for unit tests, I think I might do that next in a separate branch (finally unit tests for TLV!) before doing any further improvement in the notations.

Finally, some before/after images!

#### Before (Flat)
image

#### After (YAML)
image

#### Before (Flat)
image

#### After (YAML)
image

### CODE REVIEW

File: src/optionsdlg.ui [hunk 1]
```diff
@@ -213,6 +213,23 @@
           </item>
          </layout>
         </item>
+        <item>
+         <layout class="QFormLayout" name="notationFormLayout">
+          <item row="0" column="0">
+           <widget class="QLabel" name="notationLabel">
+            <property name="toolTip">
+             <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Changes the way the events are displayed in the main window and the detailed view. Note: you might need to 'Refresh' to see the changes take effect.&lt;br/&gt;&lt;/p&gt;&lt;p&gt;- &lt;span style=&quot; font-weight:600;&quot;&gt;Flat.&lt;/span&gt; The original format in TLV. No identation, and easy to read on non-nested events. Difficult for machines to parse.&lt;/p&gt;&lt;p&gt;- &lt;span style=&quot; font-weight:600;&quot;&gt;JSON.&lt;/span&gt; Complaint with the JSON format. Nesting is clear but long strings are difficult to read.&lt;/p&gt;&lt;p&gt;- &lt;span style=&quot; font-weight:600;&quot;&gt;YAML.&lt;/span&gt; Complaint with the YAML format. Preserves nesting and human readability while still being computer readable.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Comment (terencesoh) [inline src/optionsdlg.ui]: "Compliant with"

**Comment by luiseeo** [comment]
I don't think anyone depends or is attached to the Flat or JSON notations. But maybe someone is (?).
The YAML notation is set as the default right now, so everyone should switch automatically. We can know if someone requires the older notations if someone complains.

Another reason to not get rid of the notations just yet is that I'm sure there are some edge cases where the YAML notation is not 100% compliant. Example: Strings with ':', '\n', or '"" can be tricky. I would like to fix them after I put the unit tests in place, as it is very easy to fix something but break something else right now. (Right now, comparing the formatting between notations helps me test)
